vendors/espressif/boards/esp32/aws_tests/application_code/esp_smp_tests.c

```diff
@@ -0,0 +1,141 @@
+ */
+TEST_GROUP_RUNNER( ESP_SMP_Task_Creation )
+{
+    RUN_TEST_CASE( ESP_SMP_Task_Creation, IsMultiCore );
+    RUN_TEST_CASE( ESP_SMP_Task_Creation, CreatePinnedTaskOnCore0 );
+    RUN_TEST_CASE( ESP_SMP_Task_Creation, CreatePinnedTaskOnCore1 );
+}
+
+TEST_SETUP( ESP_SMP_Task_Creation) {}
+TEST_TEAR_DOWN( ESP_SMP_Task_Creation) {}
+
+/**
+ * @brief The response structure from report core
+ */
+typedef struct {
+    SemaphoreHandle_t done; /*< The reportCore task sets this semaphore on completion */
+	BaseType_t			xCoreID; /*< The CPU ID on which report Core ran */
+} coreReport_t;
+
+/**
+ * @brief Record the core on which this task runs, then suspend
+ */
+void reportCore( void* pvReport ) {
+    coreReport_t *  pxReport = (coreReport_t *) pvReport;
+    pxReport->xCoreID =  xPortGetCoreID();
+    if ( pxReport -> done ) {
+        if ( !xSemaphoreGive( pxReport -> done )) {
+            configPRINTF(("Error when returning semaphore."));
+        }
+    }
+    vTaskSuspend(NULL);
+}
+
+/**
+ * @brief Test whether pinned tasks run on the core to which they were assigned
+ */
+void TestCreatePinnedTask( BaseType_t xCoreID, char* taskName ) {
+    #define CORE_NOT_SET INT_MAX
+    TaskHandle_t xTaskHandle = NULL;
+    BaseType_t taskCreated = pdFALSE, taskCompleted = pdFALSE;
+    coreReport_t _coreReport = {
+        .xCoreID = CORE_NOT_SET,
+        .done = xSemaphoreCreateBinary()
+    };
+
+    coreReport_t * coreReport = &_coreReport;
+
+    if (TEST_PROTECT()) {
+
+        TEST_ASSERT_NOT_NULL_MESSAGE( coreReport -> done, "creating semaphore" );
+
+        taskCreated =  xTaskCreatePinnedToCore( reportCore,
+                                        taskName,
+                                        ESP_TASK_DEFAULT_STACK_SIZE,
+                                        coreReport,
+                                        ESP_TEST_DEFAULT_PRIORITY,
+                                        &xTaskHandle,
+                                        xCoreID);
+        TEST_ASSERT_TRUE_MESSAGE( taskCreated, "creating task" );
+
+        taskCompleted = xSemaphoreTake(  coreReport -> done , TASK_CREATION_TIMEOUT);
+
+        TEST_ASSERT_TRUE_MESSAGE( taskCompleted, "Reporting task timed out");
+        TEST_ASSERT_EQUAL_MESSAGE( coreReport -> xCoreID, xCoreID, "Core ID for pinned task");
+    }
+
+    /* Cleanup */
+    if (xTaskHandle != NULL) { 
+        vTaskDelete( xTaskHandle );
+    }
+
+    if ( coreReport -> done ) {
+        vSemaphoreDelete( coreReport -> done );
+    }
+
+}
+
+/**
+ * @brief Test that the multicore configuration flag is enabled in this build
+ */
+TEST( ESP_SMP_Task_Creation, IsMultiCore ) {
+    TEST_ASSERT_GREATER_THAN( 1,  portNUM_PROCESSORS );
+}
+
+TEST( ESP_SMP_Task_Creation, CreatePinnedTaskOnCore0 )
+{
+    TestCreatePinnedTask(0, ESP_TEST_PINNED_TASK_NAME "_0" );
+}
+
+TEST( ESP_SMP_Task_Creation, CreatePinnedTaskOnCore1 )
+{
+    TestCreatePinnedTask(1, ESP_TEST_PINNED_TASK_NAME "_1" );
+}
```

Comment:
I think it will be a good idea to verify that multiple core solution is working at a scheduler level as well.  Something in the lines of if a lower priority task can get CPU cycles when a higher priority task is using CPU cycles.

Reply:
Thanks for the feedback. We discussed offline and I added another testcase in the new revision below.